### DISCUSSION TITLE
Calling `gpiod_line_get_value()` (so v1.6, sorry!) from a signal handler

### DISCUSSION THREAD

**Discussion by RobMeades** [brgl/libgpiod#119] [Q&A] [answered]
Is it safe to call `gpiod_line_get_value()` from a signal handler?  I'm reading a set of switches that I'd like to debounce, and running a 10 ms Linux timer (the device is a Pi 5 dedicated to controlling some HW) seemed like a sensible way to do it.

I ask because my initial attempts are hitting a seg fault in `gpiod_line_is_requested()`, see below for what Valgrind says.  This may, of course, be entirely my problem I will continue to debug it, but I thought I'd ask whether I'm barking up the wrong tree...

```
==3752== Thread 14:
==3752== Invalid read of size 4
==3752==    at 0x5E13840: gpiod_line_is_requested (in /usr/lib/aarch64-linux-gnu/libgpiod.so.2.2.2)
==3752==    by 0x5E13BF3: gpiod_line_get_value_bulk (in /usr/lib/aarch64-linux-gnu/libgpiod.so.2.2.2)
==3752==    by 0x5E13D7B: gpiod_line_get_value (in /usr/lib/aarch64-linux-gnu/libgpiod.so.2.2.2)
==3752==    by 0x2414DF: gpioTickHandler(int, siginfo_t*, void*) (watchdog.cpp:1467)
==3752==    by 0x580BB9E7: ??? (in /usr/libexec/valgrind/memcheck-arm64-linux)
==3752==  Address 0x18 is not stack'd, malloc'd or (recently) free'd
```

This is Raspbian bookworm with `libgpiod` 1.6.3.

**Comment by RobMeades** [comment]
Just spotted that the Valgrind output above is referencing `libgpiod` 2.2.2, yet `pkg-config` says I have 1.6.3, so maybe things are more messed-up than I thought.

**Comment by brgl** [comment, marked as answer]
The `2.2.2` you're seeing appended to the `.so` is the ABI version, the API version is `v1.6.3`.

Answering your question: no, none of the libgpiod functions are safe to call from signal handlers. Please consider using `signalfd()`.

**Reply by RobMeades** [reply]
Ah, that's excellent information, many thanks, will use `signalfd()`.
